pkg/agent/controller/agent.go

```diff
@@ -425,11 +425,14 @@ func (a *Controller) newServiceImport(svcExport *mcsv1a1.ServiceExport) *mcsv1a1
 
 func (a *Controller) getIPsAndPortsForService(service *corev1.Service, siType mcsv1a1.ServiceImportType) (
 	[]string, []mcsv1a1.ServicePort, error) {
-	mcsPort := mcsv1a1.ServicePort{}
-	if len(service.Spec.Ports) > 0 {
-		mcsPort.Name = service.Spec.Ports[0].Name
-		mcsPort.Protocol = service.Spec.Ports[0].Protocol
-		mcsPort.Port = service.Spec.Ports[0].Port
+	mcsPorts := []mcsv1a1.ServicePort{}
```

Comment:
So we return nil if there's no ports.

```suggestion
	var mcsPorts []mcsv1a1.ServicePort
```

Reply:
```
type ServiceImportSpec struct {
	// +listType=atomic
	Ports []ServicePort `json:"ports"`
```
`ServicePort` is not `omitempty` but individual fields are. So we want it to be non nil.